### PR TITLE
refactor CocosActivity

### PR DESCRIPTION
**Pull request by bofeng-song**
Re: #  https://github.com/cocos/cocos-engine/issues/14626

### Changelog

*

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: native/cocos/platform/android/java/src/com/cocos/lib/CocosActivity.java [hunk 1]
```diff
@@ -110,100 +87,48 @@ private void setImmersiveMode() {
         }
     }
 
-    protected void initView() {
```
Comment (minggo) [inline native/cocos/platform/android/java/src/com/cocos/lib/CocosActivity.java:113]: Should keep it to keep compatible.

File: native/cocos/platform/android/java/src/com/cocos/lib/CocosActivity.java [hunk 2]
```diff
@@ -26,61 +26,38 @@
...
-        // GlobalObject.init should be initialized at first.
-        GlobalObject.init(this, this);
-
-        CocosHelper.registerBatteryLevelReceiver(this);
-        CocosHelper.init();
-        CocosAudioFocusManager.registerAudioFocusListener(this);
-        CanvasRenderingContext2DImpl.init(this);
-        this.setVolumeControlStream(AudioManager.STREAM_MUSIC);
         getWindow().setSoftInputMode(WindowManager.LayoutParams.SOFT_INPUT_ADJUST_NOTHING);
-        initView();
```
Comment (minggo) [inline native/cocos/platform/android/java/src/com/cocos/lib/CocosActivity.java:77]: Should keep it to keep compatible.

File: native/cocos/platform/android/java/src/com/cocos/lib/CocosActivity.java [hunk 3]
```diff
@@ -26,61 +26,38 @@
...
 import java.lang.reflect.Field;
-import java.util.ArrayList;
-import java.util.List;
 
 public class CocosActivity extends GameActivity {
     private static final String TAG = "CocosActivity";
-    private CocosWebViewHelper mWebViewHelper = null;
-    private CocosVideoHelper mVideoHelper = null;
+    private CocosEngine mCocosEngine;
+    private IGamePlayer mGamePlayer;
```
Comment (minggo) [inline native/cocos/platform/android/java/src/com/cocos/lib/CocosActivity.java]: I don't know why need engine and game player.

File: native/cocos/platform/android/java/src/com/cocos/lib/CocosActivity.java [hunk 4]
```diff
@@ -110,100 +85,53 @@ private void setImmersiveMode() {
...
     protected void initView() {
-        mRootLayout = findViewById(contentViewId);
-        if (mWebViewHelper == null) {
-            mWebViewHelper = new CocosWebViewHelper(mRootLayout);
-        }
-
-        if (mVideoHelper == null) {
-            mVideoHelper = new CocosVideoHelper(this, mRootLayout);
-        }
+        mCocosEngine.initView(findViewById(contentViewId));
```
Comment (dumganhar) [inline native/cocos/platform/android/java/src/com/cocos/lib/CocosActivity.java:90]: Why do we need `CocosEngine.init` and `CocosEngine.initView` both?
Should `init` do all the initialization work ?

**Pull request closed without merging**